### DISCUSSION TITLE
Is ganache strong enough for a production project?

### DISCUSSION THREAD

**Discussion by teknoxic** [ConsenSys-archive/ganache#4401] [General]
I want to use ganache for a production project, ganache will work as a side chain for my platform.
I'm just wondering if there is anything i should be aware of
such as
- Maximum blocks
- issues running for long periods of times
- ability to handle large numbers of concurrent transactions

How does ganache fair on the above and is there anything else i need to be aware of?

Thanks

**Comment by davidmurdoch** [comment]
There are features in Ganache that make it great for developers, but these features make it impossible to use as a public network. Mainly: Ganache lets users lie to it.

Also, Ganache can't communicate with other nodes. A centralized public Blockchain might as well just be a database, IMO. 🤷‍♂️

Ganache is designed to run indefinitely, though there are some RPCs, like evm_snapshot, that hold data in memory only, so you could theoretically keep calling this and consume all available memory eventually.

Check out the work the Ethereum JS team (cc @holgerd77) is doing with their client, which uses the same EVM as Ganache, but may be better suited for what you're trying to do.
